### ISSUE TITLE
Spring HttpServletRequest and HttpServletResponse

### ISSUE THREAD

**Issue by sonOfRa** [swagger-api/swagger-codegen-generators#522]
Spring supports quite a number of magic arguments in controllers: https://docs.spring.io/spring/docs/current/spring-framework-reference/web.html#mvc-ann-arguments

Among the most commonly used are probably HttpServletRequest and HttpServletResponse. While the Request can also be autowired into the controller and then accessed locally, this is not possible with HttpServletResponse. As far as I can tell, HttpServletResponse *must be* an argument to the controller method. I'm interested in adding support for this to the Spring template, but I'm not really sure of the best way to implement this. I personally would just need HttpServletResponse, so I could add a single configuration key that enables generation of that parameter in Controllers, and be done.

I'm not sure if any other "magic arguments" would need to be added to code generation, because it looks like most of them can be either autowired or acquired *through* things that can be autowired. HttpServletResponse seems to be unique in that it can *only* be acquired through the Controller arguments.

So long story short; would a PR that just adds support for HttpServletResponse to generation be welcome, or should this be done differently?

**Comment by sonOfRa** [comment]
It turns out that you *can* autowire HttpServletResponse just fine, and it was only my IDE telling me I can't, and me believing my IDE.
